src/vcf/validator.cpp

```diff
@@ -122,13 +120,52 @@ namespace ebi
     }
 
     bool is_valid_vcf_file(std::istream &input,
-                           ebi::vcf::Parser &validator,
+                           const std::string &sourceName,
+                           ValidationLevel validationLevel,
+                           Ploidy ploidy,
                            std::vector<std::unique_ptr<ebi::vcf::ReportWriter>> &outputs)
     {
+        std::vector<char> line;
+        ebi::util::readline(input, line);
+        ebi::vcf::Version version = extract_version(line);
+        std::unique_ptr<Parser> validator = build_parser(sourceName, validationLevel, version, ploidy);
+        return is_valid_vcf_file(line, input, *validator, outputs);
+    }
+
+    Version extract_version(std::vector<char> line)
+    {
+        std::string common_substring{"##fileformat=VCFv4."};
```

Comment:
Leaving the whole value (VCFv4.x) of the common substring would make this a bit more flexible. Some people are designing extensions to VCF [for proteomics](https://github.com/samtools/hts-specs/issues/160) and the fileformat declaration could change.